src/grid-list/grid-list.jsx

```diff
@@ -101,6 +101,7 @@ const GridList = React.createClass({
     const mergedRootStyles = this.mergeStyles(styles.root, style);
 
     const wrappedChildren = React.Children.map(children, (currentChild) => {
+      if (currentChild.type.displayName === 'Subheader') return currentChild;
```

Comment:
Alright, the implementation looks good: https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElement.js#L283.